controller_interface/src/controller_interface.cpp

```diff
@@ -26,24 +26,31 @@ namespace controller_interface
 {
 
 return_type
-ControllerInterface::init(const std::string & controller_name)
+ControllerInterface::on_init(const std::string & controller_name)
```

Comment:
Please delete this implementation. It's not needed because the method should be pure virtual.